File: cpp_solutions/2198. Number of Single Divisor Triplets.cpp
class Solution {
public:
    long long singleDivisorTriplet(vector<int>& nums) {
        unordered_map<int,long long int> freqs;
        long long ans = 0;
        for(int c: nums){
            if (freqs.find(c)!=freqs.end()){
                freqs[c]+=1;
            }
            else{
                freqs.insert({c,1});
            }
        }
        
        for(int x=1; x<=100;x++){
            for(int y=x;y<=100;y++){
                for (int z=y; z<=100;z++){
                    if ((freqs.find(x)!=freqs.end()) && (freqs.find(y)!=freqs.end()) && (freqs.find(z)!=freqs.end()) && valid(x,y,z)){
                        if (x==y){
                            ans+= freqs[x] * (freqs[x]-1)/2 * freqs[z];
                        }
                        else if(y==z){
                            ans+= freqs[x] * freqs[y] * (freqs[y]-1)/2;
                        }
                        else{
                            ans+= freqs[x] * freqs[y] * freqs[z];
                        }
                    }
                }
            }
        }
        
        return ans * 6;
        
    }
    bool valid(int i, int j , int k){
        int sums = i + j + k;
        
        if (sums%i==0){
            return sums%j!=0 && sums%k!=0;
        }
        else if(sums%j==0){
            return sums%k!=0;
        }
        else{
            return sums%k == 0;
        }
    }
};

File: python3_solution_set1/2198. Number of Single Divisor Triplets.py
class Solution:
    def singleDivisorTriplet(self, nums: List[int]) -> int:
        
        def valid(i,j,k):
            sums = i + j + k
            if sums%i==0:
                return sums%j!=0 and sums%k!=0
            elif sums%j == 0:
                return sums%k!=0
            
            else:
                return sums%k == 0
            
            
        comp = Counter(nums)
        ans = 0
        
        for x in range(1,101):
            for y in range(x,101):
                for z in range(y,101):
                    if x in comp and y in comp and z in comp and valid(x,y,z):
                        if x==y:
                            ans+= comp[x] * (comp[x]-1)//2 * comp[z]
                            
                        elif y==z:
                            ans+= comp[x] * (comp[y]) * (comp[y]-1)//2
                            
                        else:
                            ans+= comp[x] * comp[y] * comp[z]
                            
        return ans * 6 ## ans * 3!
        
                        
        
